invenio_stats/tasks.py

```diff
@@ -100,19 +101,22 @@ def get_bookmark(self):
                                 doc_type='{0}-bookmark'.format(self.event))
         query_bookmark.aggs.metric('latest', 'max', field='date')
         bookmark = query_bookmark.execute()[0]
-        bookmark = datetime.datetime.strptime(bookmark.date, '%Y-%m-%d')
+        # change it to doc_id_suffix
+        bookmark = datetime.datetime.strptime(bookmark.date,
+                                              self.supported_intervals[
+                                                  self.aggregation_interval])
         return bookmark.date()
 
     def set_bookmark(self):
         """Set bookmark for starting next aggregation."""
+        current_search_client.indices.flush(index='*')
+
         def _success_date():
             bookmark = {
-                'date': datetime.date.today().isoformat()
+                'date': datetime.date.today().strftime(
```

Comment:
Supposing that every document is not searchable when the events search starts, and that the processing takes some time. It would be better to take the time when the first event query was made - 1h.